python_modules/dagster/dagster/_core/definitions/definitions_loader.py

```diff
@@ -26,6 +28,20 @@ class DefinitionsLoadContext:
     def __init__(self, repository_load_data: Optional["RepositoryLoadData"] = None):
         self._repository_load_data = repository_load_data
 
+    @cached_property
```

Comment:
we should have an expliict state where this is available (when you are reconstructing an already-constructed code location), and it should throw if you try to access it when it is in an initial state where the code location doesn't exist yet. This is a bootstrapping problem.

Reply:
Agree, this will require plumbing a "context type" or something into the `DefinitionsLoadContext` which can happen in another PR.

Reply:
Define "next iteration". I think that context type or context state should be in an immediate follow up

Reply:
Have now implemented in a downstack PR: https://github.com/dagster-io/dagster/pull/24490